savant/utils/image.py

```diff
@@ -0,0 +1,512 @@
+            assert (
+                self.height == image.height
+            ), f'Images have different width {self.height} != {image.height}'
+            res_cols = self.width + image.width
+            res_image = cv2.cuda.GpuMat(
+                size=(res_cols, self.height), type=self._gpu_image.type()
+            )
+            self.gpu_mat.copyTo(
+                stream=self._cuda_stream, dst=res_image.colRange(0, self.width)
+            )
+            image.gpu_mat.copyTo(
+                stream=self._cuda_stream, dst=res_image.colRange(self.width, res_cols)
+            )
+            return GPUImage(image=res_image, cuda_stream=self._cuda_stream)
+        else:
+            raise ValueError(f'Unknown axis {axis}')
+
+    def paste(self, image: 'GPUImage', point: Tuple[int, int]):
+        """Pastes image on current image.
+
+        :param image: image to paste
+        :param point: point to paste (x, y)
+        """
+
+        frame_roi = self._gpu_image.colRange(point[0], point[0] + image.width).rowRange(
+            point[1], point[1] + image.height
+        )
+        image.gpu_mat.copyTo(stream=self._cuda_stream, dst=frame_roi)
+
+    def rotate(
+        self, angle: float, bbox: Optional[RBBox] = None
+    ) -> Tuple['GPUImage', RBBox]:
+        """Rotates image on angle. If.
+
+        :param angle: angle to rotate in degrees
+        :param bbox: bounding box on image
+        :return: rotated image
+        """
+
+        rotation_matrix, resolution = get_rotation_matrix(
+            self, angle, (bbox.x_center, bbox.y_center)
+        )
+
+        res = cv2.cuda.warpAffine(
+            src=self._gpu_image,
+            M=rotation_matrix,
+            dsize=resolution,
+            stream=self._cuda_stream,
+        )
+
+        if bbox is not None:
+            return GPUImage(image=res, cuda_stream=self._cuda_stream), RBBox(
+                x_center=resolution[0] / 2,
+                y_center=resolution[1] / 2,
+                width=bbox.width,
+                height=bbox.height,
+                angle=bbox.angle - angle,
+            )
+        return GPUImage(image=res, cuda_stream=self._cuda_stream), RBBox(
+            x_center=resolution[0] / 2,
+            y_center=resolution[1] / 2,
+            width=self.width,
+            height=self.height,
+            angle=angle,
+        )
+
+    def resize(
+        self,
+        resolution: Tuple[int, int],
+        method: str = 'fit',
+        interpolation: int = cv2.INTER_LINEAR,
+    ) -> 'GPUImage':
+        """Resizes image to resolution.
+
+        :param resolution: resolution to resize [width, height]
+        :param method: method to resize one of ['fit', 'scale']
+        :param interpolation: interpolation method.
+            Oone of [cv2.INTER_NEAREST, cv2.INTER_AREA, cv2.INTER_LINEAR,
+            cv2.INTER_CUBIC, cv2.INTER_LANCZOS4]
+        :return: resized image
+        """
+        new_resolution = tuple(map(int, resolution))
```

Comment:
But the resolution is already annotated to be a tuple of ints. Do we need to try to convert? If so, then the annotation needs to include other expected types.

Reply:
removed